src/adapter/src/catalog/builtin_table_updates.rs

```diff
@@ -571,9 +572,42 @@ impl CatalogState {
             }
         }
 
+        // Use initial lcw so that we can tell apart default from non-existent windows.
+        if let Some(cw) = entry.item().initial_logical_compaction_window() {
+            updates.push(self.pack_history_retention_strategy_update(id, cw, diff));
+            // Propagate subsource changes.
+            for (cw, mut ids) in self.source_compaction_windows([id]) {
+                // Id already accounted for above.
+                ids.remove(&id);
+                for sub_id in ids {
+                    updates.push(self.pack_history_retention_strategy_update(sub_id, cw, diff));
+                }
+            }
+        }
+
         updates
     }
 
+    fn pack_history_retention_strategy_update(
+        &self,
+        id: GlobalId,
+        cw: CompactionWindow,
+        diff: Diff,
+    ) -> BuiltinTableUpdate {
+        let cw: u64 = cw.comparable_timestamp().into();
+        let cw = Jsonb::from_serde_json(serde_json::Value::Number(serde_json::Number::from(cw)))
+            .expect("must serialize");
+        BuiltinTableUpdate {
+            id: self.resolve_builtin_table(&MZ_HISTORY_RETENTION_STRATEGIES),
+            row: Row::pack_slice(&[
+                Datum::String(&id.to_string()),
+                Datum::String("FOR"),
```

Comment:
Can we get a comment about why we're hard-coding `"FOR"`? It looks a bit odd.

Reply:
Done.